utility commands (had to move some branches around)

arnokeesman opened the pull request:
feed, heal, near, repair, night, suicide, extinguish
closes: #38
related: #244

```
/feed [player]
enable_feed: true
essentialcommands.feed.self 2
essentialcommands.feed.others 2

/heal [player]
enable_heal: true
essentialcommands.heal.self 2
essentialcommands.heal.others 2

/near [range] [player]
enable_near: true
near_command_default_range: 200
near_command_max_range: 200
essentialcommands.near.self 2
essentialcommands.near.others 2

/repair [player]
enable_repair: true
essentialcommands.repair.self 2
essentialcommands.repair.others 2

/night
enable_night: true
essentialcommands.night 2

/suicide
enable_suicide: true
essentialcommands.suicide 0

/ext(inguish) [player]
enable_extinguish: true
essentialcommands.extinguish.self 2
essentialcommands.extinguish.others 2
```
utility commands moved to commands.utility package didn't feel like /near fit in the utility package

arnokeesman committed:
utility commands

feed, heal, near, repair, night, suicide, extinguish
closes: #38
related: #244

/feed [player]
enable_feed: true
essentialcommands.feed.self 2
essentialcommands.feed.others 2

/heal [player]
enable_heal: true
essentialcommands.heal.self 2
essentialcommands.heal.others 2

/near [range] [player]
enable_near: true
near_command_default_range: 200
near_command_max_range: 200
essentialcommands.near.self 2
essentialcommands.near.others 2

/repair [player]
enable_repair: true
essentialcommands.repair.self 2
essentialcommands.repair.others 2

/night
enable_night: true
essentialcommands.night 2

/suicide
enable_suicide: true
essentialcommands.suicide 0

/ext(inguish) [player]
enable_extinguish: true
essentialcommands.extinguish.self 2
essentialcommands.extinguish.others 2

utility commands moved to commands.utility package
didn't feel like /near fit in the utility package

John-Paul-R commented on src/main/java/com/fibermc/essentialcommands/commands/utility/NightCommand.java:
```diff
@@ -0,0 +1,29 @@
+            return -1;
+        }
+        long time = world.getTimeOfDay();
+        long timeToDay = 13000L - time % 24000L;
+
+        world.setTimeOfDay(time + timeToDay);
```
```suggestion
        long timeToNight = 13000L - time % 24000L;

        world.setTimeOfDay(time + timeToNight);
```

arnokeesman replied:
nice catch, this change must've gotten lost when i moved these commands to files

John-Paul-R commented on src/main/java/com/fibermc/essentialcommands/commands/NearCommand.java:
```diff
@@ -0,0 +1,67 @@
+        return exec(senderPlayerData, targetPlayer, range);
+    }
+
+    public static int exec(PlayerData senderPlayerData, ServerPlayerEntity targetPlayer, int range) {
+        Vec3d basePos = targetPlayer.getPos();
+
+        MutableText results = Text.empty();
+        boolean matchesFound = false;
+        for (PlayerEntity player : targetPlayer.getWorld().getPlayers()) {
+            if (Objects.equals(targetPlayer.getUuid(), player.getUuid())) continue;
+            if (!basePos.isInRange(player.getPos(), range)) continue;
+            if (EssentialCommands.VANISH_PRESENT && !VanishAPI.canSeePlayer((ServerPlayerEntity) player, senderPlayerData.getPlayer())) continue;
+
+            if (matchesFound) results.append(", ");
+            results.append(player.getDisplayName());
+            matchesFound = true;
+        }
+
+        if (!matchesFound) senderPlayerData.sendCommandFeedback("cmd.near.feedback.empty");
+        else senderPlayerData.sendCommandFeedback("cmd.near.feedback.list", results);
+
+        return 1;
+    }
```
This is a "do it if you like it" suggestion. What have is near-certainly faster, though.

```suggestion
    public static int exec(PlayerData senderPlayerData, ServerPlayerEntity targetPlayer, int range) {
        Vec3d basePos = targetPlayer.getPos();

        var players = targetPlayer.getWorld().getPlayers().stream()
            .filter(player ->
                targetPlayer.getUuid() != player.getUuid()
                && !basePos.isInRange(player.getPos(), range)
                && EssentialCommands.VANISH_PRESENT && !VanishAPI.canSeePlayer((ServerPlayerEntity) player, senderPlayerData.getPlayer())
            )
            .map(PlayerEntity::getDisplayName)
            .toList();

        if (players.isEmpty()) senderPlayerData.sendCommandFeedback("cmd.near.feedback.empty");
        else senderPlayerData.sendCommandFeedback("cmd.near.feedback.list", TextUtil.join(players, Text.literal(", ")));

        return SINGLE_SUCCESS;
    }
```

John-Paul-R commented on src/main/java/com/fibermc/essentialcommands/commands/NearCommand.java:
```diff
@@ -0,0 +1,67 @@
+import net.minecraft.text.Text;
+import net.minecraft.util.math.Vec3d;
+
+public class NearCommand implements Command<ServerCommandSource> {
```
should this be in the `utility` package since it doesn't rely on any persistent mod state? (Aside from Vanish, which I think is more of an integration than part of the feature's primary purpose, if that makes sense)

arnokeesman replied:
yeah sounds good

arnokeesman committed:
PR comments
fix variable name
move from for loop to stream filtering (and fix logic (was good before))
moved near and gametime commands to utility package
